docs/DeveloperGuide.md

```diff
@@ -238,33 +238,84 @@ _{more aspects and alternatives to be added}_
 
 #### Implementation
 
-The clone feature creates a copy of a person in the addressbook while only adding a number at the end of the name and maintaining all other variables. 
+The `clone` feature creates a copy of a person in the address book while either adding a suffix at the end of the name 
+of the cloned contact or, if the contact name has a pre-existing suffix, it increments that suffix by one.
+
+The clone mechanism first checks to see if the index provided is valid. If a person exists at the index, it then checks
+the suffix of the person in question.
+
+Clone implements the following operations:
+* `CloneCommand#execute`
+* `CloneCommand#equals`
+* `CloneCommand#toString`
+* `CloneCommand#clonePerson`
+* `CloneCommand#splitStringAtLastSpace`
+
+These operations make use of other operations exposed in the `Model` interface, which are:
+* `Model#getFIlteredPersonList()`
+* `Model#addPerson(Person)`
+* `Model#storePreviousUndoableCommand(String)`
+* `Model#resetRedoableStateList()`
+* `Model#resetUndoableStateList()`
+* `Model#removeRedoCommands()`
 
 Given below is an example usage scenario and how the clone mechanism behaves at each step.
 
-Step 1. The user executes "list" to see what Persons are available in the address book
+Step 1. The user launches the application for the first time. 
+
+Step 2. The user executes `list` to see what Persons are available in the address book
```

Comment:
```suggestion
Step 2. The user executes `list` to see what Persons are available in the address book.
```

Perhaps there should be a full-stop at the end here? Just to preserve consistency with Step 1 and Step 3.